zng/resolver/context.go

```diff
@@ -37,84 +38,162 @@ func NewContext() *Context {
 
 // Row is a structure used to organize the generic type table
 // into type-specific subcomponents by category name.
-type Row struct {
-	Category string      `json:"name"`
-	Type     interface{} `json:"type"`
+type Alias struct {
+	Name string
+	Id   int
+}
+
+const (
+	TypeDefRecord = 0x80
+	TypeDefArray  = 0x81
+	TypeDefSet    = 0x82
+	TypeDefAlias  = 0x83
+)
+
+// XXX for now, we marshal into this data structure to represent the entire
+// type table.  After we update the ZNG implementation to use typedefs, we
+// will write a context table as BZNG.
+type TypeDef struct {
+	Id      int
+	Code    int
+	Aliases []Alias
+}
+
+func setTypeDef(id int, t *zng.TypeSet) TypeDef {
+	return TypeDef{
+		Id:      id,
+		Code:    TypeDefSet,
+		Aliases: []Alias{Alias{"set", t.InnerType.Id()}},
```

Comment:
Nit: Just say `[]Alias{{"set", t.InnerType.Id()}},`.

Reply:
cool, learn something new every day

Reply:
btw, all this marshal stuff is boiler plate that I will throw out.  I needed the wedge while I finish the update to conform with the spec

Reply:
though some of these data types may very well stick around